src/main/java/org/opentripplanner/updater/transportation_network_company/OAuthToken.java

```diff
@@ -0,0 +1,58 @@
+package org.opentripplanner.updater.transportation_network_company;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.time.Instant;
+import java.util.Date;
+
+/**
+ * Holds an OAuth access token and its expiration time for querying ride-hail APIs.
+ */
+public class OAuthToken {
+    public final String value;
+    private Date tokenExpirationTime;
+
+    private OAuthToken() {
+        value = null;
+    }
+
+    public static OAuthToken blank() {
+        return new OAuthToken();
+    }
+
+    public OAuthToken(HttpURLConnection connection) {
+        // send request and parse response
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Can be static.

Reply:
Updated in https://github.com/ibi-group/OpenTripPlanner/pull/115/commits/4b7970237b804bd0c64a14a9b7acda08302567be.